mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -1422,3 +1424,261 @@ def _auto_indent(repr_str, indent):
         repr_str += ' ' * indent + 'prev_voxel_generator=\n'
         repr_str += f'{_auto_indent(repr(self.prev_voxel_generator), 8)})'
         return repr_str
+
+
+@PIPELINES.register_module()
+class AffineResize(object):
+    """Get the affine transform matrixs to the target size.
+
+    Different from RandomAffine in MMDetection, this class can calculate
+    the affine transform matrixs while resizing the input image to a fixed
+    size. The affine transform matrixs include:
+    1) matrix transforming original image to the network input image
+        size.
+    2) matrix transforming original image to the network output feature
+        map size.
+
+    Args:
+        img_scale (tuple): Images scales for resizing.
+        down_ratio (int): The down ratio of feature map.
+            Actually the arg should be >= 1.
+        bbox_clip_border (bool, optional): Whether clip the objects
+            outside the border of the image. Defaults to True.
+    """
+
+    def __init__(self, img_scale=None, down_ratio=None, bbox_clip_border=True):
```

Comment:
Are all the parameters optional? I guess at least img_scale is a necessary argument?